### PR TITLE
xorgproto: new package

### LINKED ISSUE #41688: Newer versions of freedesktop.org/Xorg inputproto and fixesproto not released separately.
### Summary

Newer versions of freedesktop.org/Xorg inputproto and fixesproto are not released separately anymore.

### Rationale

Xorg has started releasing the entire protocol set as a single distribution. Spack is not setup for this right now. Since newer versions of the protocols are not being released independently, we cannot update Xorg libraries that depend on those newer protocol versions until we figure out how to transition from individual protocols to a single xorgproto package.

### Description

One possible solution is in https://github.com/spack/spack/pull/36238 but expands the number of virtual packages more than desired.

### Additional information

_No response_

### General information

- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by teaguesterling**
Breaking out the xorgproto package that replaces the individual x proto files from #44722. This package has lots of virtual providers, so the issues that appeared in #45485 should be considered and evaluated separately.

Potentially, this could also deprecate the other proto files over time, as that is the direction that X seems to be taking.

Given the number of packages that the xorgproto package subsumes, I wonder if it would make more sense to simply update the packages that depend on those various protos to use xorgproto. That would cut down on the number of "providers" that we are adding.

Noting that @wdconinc had a previous PR in #36238 that has some important things to consider before merging.

### CODE REVIEW

**Comment by wdconinc** [comment]
I started something similar a long time ago, https://github.com/spack/spack/pull/36238. Looking at that now, I did get a few more provides for other protocols too (and `pkgconfig` as a build dep).

I found the virtuals provider approach the most straightforward, but at that time it more than doubled the number of virtual packages so it was unwieldy. I think that it may indeed make more sense to approach this without providers and with link dependencies on `xorgproto`.

**Comment by teaguesterling** [comment]
> I found the virtuals provider approach the most straightforward, but at that time it more than doubled the number of virtual packages so it was unwieldy. I think that it may indeed make more sense to approach this without providers and with link dependencies on `xorgproto`.

I agree that it would make more sense to just link to `xorgproto`. It will make this PR much larger, but I'm more comfortable with that than adding a bunch of virtual packages just for header files at build time.

**Comment by teaguesterling** [comment]
@wdconinc, your version also put a lot more thought into this than I did. I suspect there are at least a few things we should take from there to make this more resilient/complete. My aim with this was only to get turbovnc to build, so I'm sure I've missed something.

**Comment by wdconinc** [comment]
I recall now what the main problem was with the providers: the semver versions of the individual protocols are different from the calver versions of the `xorg-proto` bundles. The Xorg packages do impose protocol version minimum versions, so these need to be propagated. That's why the list of providers becomes quite unwieldy to manage.

In that sense, I wonder if we can avoid the provider approach with something like this (using `fixesproto` where a new protocol version is only in the bundle):
```python
class Fixesproto(AutotoolsPackage, XorgPackage, BundlePackage):
    """X Fixes Extension.

    The extension makes changes to many areas of the protocol to resolve
    issues raised by application interaction with core protocol mechanisms
    that cannot be adequately worked around on the client side of the wire."""

    homepage = "https://cgit.freedesktop.org/xorg/proto/fixesproto"
    xorg_mirror_path = "proto/fixesproto-5.0.tar.gz"

    version("6.0")
    version("5.0", sha256="67865a0e3cdc7dec1fd676f0927f7011ad4036c18eb320a2b41dbd56282f33b8")
    
    build_system(
        conditional(autotools, when="@:5"),
        conditional(bundle, when="@6:"),
        default=bundle,
    )

    with when("build_system=autotols"):
        depends_on("pkgconfig", type="build")
        depends_on("util-macros", type="build")

    with when("build_system=bundle"):
        depends_on("xorg-proto@2021.4:", when="@6.0:")
```

*Edit* Note: this is entirely untested. I have no reason to expect that you would not be able to use a bundle package as a multiple build system option, but it is not used in any other package as far as I can tell.

**Comment by wdconinc** [comment]
> It will make this PR much larger

I think the providers part is what makes it harder to get this into develop. Without the providers, this could get merged easily as a new package, and then the next step (in a different PR) could be to add the necessary support in the individual protocols, piecemeal and as needed. Or to just change each protocol to a bundle package that depends on `xorg-proto`.

**Comment by wdconinc** [comment]
So, I got a tested working case without providers (sorry for the length of output in email notification). This is with `xorg-proto` instead of `xorgproto` as you have called it (I prefer your package name).

<details><summary><verbatim>${SPACK_ROOT}/var/spack/repos/builtin/packages/fixesproto/package.py</verbatim></summary>
This requires `has_code = True` because otherwise spack doesn't download for the `AutotoolsPackage`. It also requires a `touch ${SPACK_ROOT}/var/spack/repos/builtin/packages/fixesproto/empty` because we cannot set `has_code` as an individual version attribute.
<pre>
# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
# Spack Project Developers. See the top-level COPYRIGHT file for details.
#
# SPDX-License-Identifier: (Apache-2.0 OR MIT)

from spack.package import *
from spack.build_systems import autotools, bundle


class Fixesproto(AutotoolsPackage, XorgPackage, BundlePackage):
    """X Fixes Extension.

    The extension makes changes to many areas of the protocol to resolve
    issues raised by application interaction with core protocol mechanisms
    that cannot be adequately worked around on the client side of the wire."""

    homepage = "https://cgit.freedesktop.org/xorg/proto/fixesproto"
    xorg_mirror_path = "proto/fixesproto-5.0.tar.gz"

    _pkg_dir = spack.repo.PATH.dirname_for_package_name("fixesproto")
    _sha_empty = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"

    version("6.0", url=f"file://{_pkg_dir}/empty", sha256=_sha_empty, expand=False)
    version("5.0", sha256="67865a0e3cdc7dec1fd676f0927f7011ad4036c18eb320a2b41dbd56282f33b8")

    has_code = True

    build_system(
        conditional("autotools", when="@:5"),
        conditional("bundle", when="@6:"),
        default="bundle",
    )

    with when("build_system=autotools"):
        depends_on("pkgconfig", type="build")
        depends_on("util-macros", type="build")

    with when("build_system=bundle"):
        depends_on("xorg-proto@2021.4:", when="@6.0:")
</pre>
</details>

<details><summary><verbatim>${SPACK_ROOT}/var/spack/repos/builtin/packages/libxfixes/package.py</verbatim></summary>
This is just adding the new version that requires `fixesproto >= 6`.
<pre>
# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
# Spack Project Developers. See the top-level COPYRIGHT file for details.
#
# SPDX-License-Identifier: (Apache-2.0 OR MIT)

from spack.package import *


class Libxfixes(AutotoolsPackage, XorgPackage):
    """This package contains header files and documentation for the XFIXES
    extension.  Library and server implementations are separate."""

    homepage = "https://gitlab.freedesktop.org/xorg/lib/libXfixes"
    xorg_mirror_path = "lib/libXfixes-5.0.2.tar.gz"

    license("MIT")

    maintainers("wdconinc")

    # Newer versions are blocked by https://github.com/spack/spack/issues/41688
    version("6.0.1", sha256="e69eaa321173c748ba6e2f15c7cf8da87f911d3ea1b6af4b547974aef6366bec")
    version("6.0.0", sha256="82045da5625350838390c9440598b90d69c882c324ca92f73af9f0e992cb57c7")
    version("5.0.3", sha256="9ab6c13590658501ce4bd965a8a5d32ba4d8b3bb39a5a5bc9901edffc5666570")
    version("5.0.2", sha256="ad8df1ecf3324512b80ed12a9ca07556e561b14256d94216e67a68345b23c981")

    depends_on("c", type="build")  # generated

    depends_on("libx11@1.6:")

    depends_on("xproto")
    depends_on("fixesproto@5.0:", when="@5")
    depends_on("fixesproto@6.0:", when="@6")
    depends_on("xextproto")
    depends_on("pkgconfig", type="build")
    depends_on("util-macros", type="build")
</pre>
</details>

<details><summary>Test environment:</summary>
<pre>
# This is a Spack Environment file.
#
# It describes a set of packages to be installed, along with
# configuration settings.
spack:
  # add package specs to the `specs` list
  specs:
  - libxfixes@6.0.1 ^fixesproto@6
  - libxfixes@5.0.3 ^fixesproto@5
  view: false
  concretizer:
    unify: false
  packages:
    perl:
      externals:
      - spec: perl@5.38.2~cpanm+opcode+open+shared+threads
        prefix: /usr
      buildable: false
    python:
      externals:
      - spec: python@3.12.3+bz2+crypt+ctypes+dbm+lzma~nis+pyexpat+pythoncmd+readline+sqlite3+ssl+tix+tkinter+uuid+zlib
        prefix: /usr
      buildable: false
</pre>
</details>

<details><summary>Concretizes as:</summary>
<pre>
$ spack concretize --fresh --force 
==> Starting concretization pool with 2 processes
==> Environment concretized in 4.01 seconds
==> Concretized 2 specs:
 -   7w2o4vf  libxfixes@5.0.3%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
 -   l62jnr2      ^fixesproto@5.0%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  rjdzeos      ^gcc-runtime@13.2.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[e]  o2zy2ye      ^glibc@2.39%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  gmcvx5j      ^gmake@4.4.1%gcc@13.2.0~guile build_system=generic arch=linux-ubuntu24.04-skylake
[+]  7g3p5dn      ^libx11@1.8.9%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  q4lbh2p          ^inputproto@2.3.2%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  le64ocx          ^kbproto@1.0.7%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  5ek5pwq          ^libxcb@1.17.0%gcc@13.2.0~use_spack_interpreter build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  qjmtp6v              ^libpthread-stubs@0.5%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  tg5q6bq              ^libxau@1.0.11%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  pruxtnf              ^libxdmcp@1.1.5%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  agqymek                  ^autoconf@2.72%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  5sadx7z                      ^m4@1.4.19%gcc@13.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu24.04-skylake
[+]  vhochvw                          ^diffutils@3.10%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  n35axwd                              ^libiconv@1.17%gcc@13.2.0 build_system=autotools libs=shared,static arch=linux-ubuntu24.04-skylake
[+]  ywje6er                          ^libsigsegv@2.14%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  vb76tg3                  ^libbsd@0.12.2%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  j5scuyb                      ^libmd@1.1.0%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  soxsf6x                          ^automake@1.16.5%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[e]  ddhhm7a              ^python@3.12.3%gcc@13.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl+tix+tkinter+uuid+zlib build_system=generic arch=linux-ubuntu24.04-skylake
[+]  hjkhzva              ^xcb-proto@1.17.0%gcc@13.2.0~use_spack_interpreter build_system=autotools arch=linux-ubuntu24.04-skylake
[e]  ivomb6c          ^perl@5.38.2%gcc@13.2.0~cpanm+opcode+open+shared+threads build_system=generic arch=linux-ubuntu24.04-skylake
[+]  h5svx5y          ^xtrans@1.5.0%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  g5u7clr      ^pkgconf@2.2.0%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  igvtyhw      ^util-macros@1.20.1%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  l65cme6      ^xextproto@7.3.0%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
[+]  okrjuyq      ^xproto@7.0.31%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
 -   cf5w7c4  libxfixes@6.0.1%gcc@13.2.0 build_system=autotools arch=linux-ubuntu24.04-skylake
 -   il5gyog      ^fixesproto@6.0%gcc@13.2.0 build_system=bundle arch=linux-ubuntu24.04-skylake
[+]  ela3dne          ^xorg-proto@2024.1%gcc@13.2.0~legacy~strip build_system=meson buildtype=release default_library=shared arch=linux-ubuntu24.04-skylake
[+]  dfxnwgc              ^meson@1.5.1%gcc@13.2.0 build_system=python_pip patches=0f0b1bd arch=linux-ubuntu24.04-skylake
[+]  inywvgn                  ^py-pip@24.1.1%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[+]  a6z3yvl                  ^py-setuptools@70.2.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[+]  cdwu4y2                  ^py-wheel@0.43.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[+]  xaubztq                  ^python-venv@1.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
[+]  bdvztpj              ^ninja@1.12.1%gcc@13.2.0+re2c build_system=generic arch=linux-ubuntu24.04-skylake
[+]  yfijrfj                  ^re2c@3.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu24.04-skylake
</pre>
</details>

<details><summary>Build output (relevant bits)</summary>
<pre>
==> Installing fixesproto-5.0-l62jnr2lss7tyg7hvsikyhgd3vrsfhjb [18/28]
==> No binary for fixesproto-5.0-l62jnr2lss7tyg7hvsikyhgd3vrsfhjb found: installing from source
==> Using cached archive: /opt/spack/cache/_source-cache/archive/67/67865a0e3cdc7dec1fd676f0927f7011ad4036c18eb320a2b41dbd56282f33b8.tar.gz
==> No patches needed for fixesproto
==> fixesproto: Executing phase: 'autoreconf'
==> fixesproto: Executing phase: 'configure'
==> fixesproto: Executing phase: 'build'
==> fixesproto: Executing phase: 'install'
==> fixesproto: Successfully installed fixesproto-5.0-l62jnr2lss7tyg7hvsikyhgd3vrsfhjb
  Stage: 0.01s.  Autoreconf: 0.00s.  Configure: 1.60s.  Build: 0.00s.  Install: 0.01s.  Post-install: 0.07s.  Total: 1.72s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/fixesproto-5.0-l62jnr2lss7tyg7hvsikyhgd3vrsfhjb
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libxau-1.0.11-tg5q6bqkf2cghjneprsqeuu5dayi5ljv
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libbsd-0.12.2-vb76tg3k3kgnwb3vtwddrofig25ktuy2
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/meson-1.5.1-dfxnwgcpdi6ooxobprv7dbwauw3l4do4
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libxdmcp-1.1.5-pruxtnfr4bsnhi6cvg4ufk7ryjdtp2i6
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/xorg-proto-2024.1-ela3dne6inknw3dtpczaptuyqvrcrn5b
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libxcb-1.17.0-5ek5pwqaieirklt2c3upgssbmacpwhzb
==> Installing fixesproto-6.0-il5gyogqijwykbeq7zln47bruwhkxzfa [25/28]
==> No binary for fixesproto-6.0-il5gyogqijwykbeq7zln47bruwhkxzfa found: installing from source
==> Using cached archive: /opt/spack/cache/_source-cache/archive/e3/e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
==> No patches needed for fixesproto
==> fixesproto: Executing phase: 'install'
==> fixesproto: Successfully installed fixesproto-6.0-il5gyogqijwykbeq7zln47bruwhkxzfa
  Stage: 0.00s.  Install: 0.00s.  Post-install: 0.15s.  Total: 0.18s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/fixesproto-6.0-il5gyogqijwykbeq7zln47bruwhkxzfa
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libx11-1.8.9-7g3p5dn3ehnpenvxbf3mklt6scasyixa
==> Installing libxfixes-6.0.1-cf5w7c4rqkgpppxc2iwh5dendhucpvr3 [27/28]
==> No binary for libxfixes-6.0.1-cf5w7c4rqkgpppxc2iwh5dendhucpvr3 found: installing from source
==> Using cached archive: /opt/spack/cache/_source-cache/archive/e6/e69eaa321173c748ba6e2f15c7cf8da87f911d3ea1b6af4b547974aef6366bec.tar.gz
==> No patches needed for libxfixes
==> libxfixes: Executing phase: 'autoreconf'
==> libxfixes: Executing phase: 'configure'
==> libxfixes: Executing phase: 'build'
==> libxfixes: Executing phase: 'install'
==> libxfixes: Successfully installed libxfixes-6.0.1-cf5w7c4rqkgpppxc2iwh5dendhucpvr3
  Stage: 0.01s.  Autoreconf: 0.00s.  Configure: 2.68s.  Build: 0.25s.  Install: 0.12s.  Post-install: 0.17s.  Total: 3.31s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libxfixes-6.0.1-cf5w7c4rqkgpppxc2iwh5dendhucpvr3
==> Installing libxfixes-5.0.3-7w2o4vf37osstbstf5oxfkt3ruckbojw [28/28]
==> No binary for libxfixes-5.0.3-7w2o4vf37osstbstf5oxfkt3ruckbojw found: installing from source
==> Using cached archive: /opt/spack/cache/_source-cache/archive/9a/9ab6c13590658501ce4bd965a8a5d32ba4d8b3bb39a5a5bc9901edffc5666570.tar.gz
==> No patches needed for libxfixes
==> libxfixes: Executing phase: 'autoreconf'
==> libxfixes: Executing phase: 'configure'
==> libxfixes: Executing phase: 'build'
==> libxfixes: Executing phase: 'install'
==> libxfixes: Successfully installed libxfixes-5.0.3-7w2o4vf37osstbstf5oxfkt3ruckbojw
  Stage: 0.01s.  Autoreconf: 0.00s.  Configure: 2.98s.  Build: 0.17s.  Install: 0.09s.  Post-install: 0.10s.  Total: 3.41s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/libxfixes-5.0.3-7w2o4vf37osstbstf5oxfkt3ruckbojw
</pre>
</details>

*Edit*: fixed some conceptual errors, typos, updated output.

**Comment by wdconinc** [comment]
Updated comment above. It seems like combining bundle versions without code and autotools versions which have code requires us (for now) to provide a fake archive file. We probably should have `has_code` be as a version-level overridable attribute, not just a package-level attribute.

**Comment by teaguesterling** [comment]
> Updated comment above. It seems like combining bundle versions without code and autotools versions which have code requires us (for now) to provide a fake archive file. We probably should have `has_code` be as a version-level overridable attribute, not just a package-level attribute.

This is great! Do you know if we can use `has_code` as a property in the meantime or does it need to be a class level properly? I can start going through the packages and using this pattern instead of the virtual packages. This would also work for the situation in #45571.

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py]: Based on discussion, let's avoid the providers.
```suggestion
```

File: etc/spack/defaults/packages.yaml
Comment (wdconinc) [inline etc/spack/defaults/packages.yaml]:
```suggestion
      blas: [openblas, amdblis]
      D: [ldc]
      daal: [intel-oneapi-daal]
      elf: [elfutils]
      fftw-api: [fftw, amdfftw]
      flame: [libflame, amdlibflame]
```

File: etc/spack/defaults/packages.yaml
Comment (wdconinc) [inline etc/spack/defaults/packages.yaml]:
```suggestion
      qmake: [qt-base, qt]
      rpc: [libtirpc]
      scalapack: [netlib-scalapack, amdscalapack]
      sycl: [hipsycl]
      szip: [libaec, libszip]
      tbb: [intel-tbb]
      unwind: [libunwind]
      uuid: [util-linux-uuid, libuuid]
      wasi-sdk: [wasi-sdk-prebuilt]
```

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py:21]: Per https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/blob/master/meson.build?ref_type=heads#L23
```suggestion
    depends_on("meson@0.56:")
    depends_on("util-macros", type="build")
```

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py]:
```suggestion
class Xorgproto(MesonPackage, XorgPackage):
```

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py]: XorgPackage has mirrors, e.g.
```suggestion
    xorg_mirror_path = "proto/xorgproto-2024.1.tar.gz"
```
Mirrors only publish gz (for all version) and bz2 (older version) or xz (newer versions), so sticking to gz is most portable.

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py]: Iff switching to gz in xorg_mirror_path:
```suggestion
    version("2024.1", sha256="4f6b9b4faf91e5df8265b71843a91fc73dc895be6210c84117a996545df296ce")
```

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py:19]: There is one option, `legacy`, https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/blob/master/meson_options.txt?ref_type=heads, which avoids installing some rarely used protocols, https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/blob/master/meson.build?ref_type=heads#L88.

Comment (teaguesterling) [reply var/spack/repos/builtin/packages/xorgproto/package.py:19]: We should add this as a variant, yeah?

**Comment by teaguesterling** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [reply var/spack/repos/builtin/packages/xorgproto/package.py:19]: I think that would be preferred, but we could also just by default run with `legacy=OFF` (probably explicitly in `meson_args`).

We'd have to see what we would actually lose without the legacy protocols... It doesn't look like we would be restricting anything much.
```
$ for i in 'evieproto' 'fontcacheproto' 'lg3dproto' 'printproto' 'trapproto' 'windowswmproto' 'xcalibrateproto' 'xf86miscproto' 'xf86rushproto' 'xproxymngproto' ; do echo -n "$i: " ; spack dependents -t $i ; done
evieproto: No dependents
fontcacheproto: libxfontcache
lg3dproto: No dependents
printproto: grace  libxp  libxprintapputil  libxprintutil  xphelloworld  xplsprinters  xprehashprinterlist
trapproto: libxtrap  xtrap
windowswmproto: libwindowswm
xcalibrateproto: No dependents
xf86miscproto: libxxf86misc
xf86rushproto: No dependents
xproxymngproto: No dependents
```

Comment (wdconinc) [reply var/spack/repos/builtin/packages/xorgproto/package.py:19]: Let's just drop these legacy protocols here.

File: var/spack/repos/builtin/packages/xorgproto/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/xorgproto/package.py]:
```suggestion
    depends_on("meson@0.56:", type="build")
```

**Review by wdconinc — APPROVED** [review]

**Pull request merged**
